### ISSUE TITLE
Do basic Dojo build for UI widget plugins (reusing davinci.gridx's customBuildCallbacks.xml file)

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#2501]
For #2390 ("Do custom build for GridX"), I created a `customBuildCallbacks.xml` file that builds one layer for the gridx library and another layer for the gridx helpers, gridx wizard, etc. I took care to make `customBuildCallbacks.xml` generic (e.g., it contains no hardcoded references to "gridx"). So, I think it could be applied to the other UI widget plugins with relatively little effort to build both their libraries and helpers.

Doing so is not as ambitious as the proposal in #1152 ("Script for building JS library metadata/helpers") in that it doesn't do anything with the json metadata files, but comes (I think) at a relatively small cost. And, I believe this could be a a stepping stone to a full implementation of #1152.

**Comment by peller** [comment]
The problem isn't so much how to build, but what to build.  As we discussed, building unrelated code into one giant layer would be a mistake.  gridx, at least without plugin options, is a great candidate for a layer.  Some helpers may involve many dependencies and also be good candidates for a build.  Others may be one-offs, where lazy-loading or pre-fetching may be a better option.  Also, once we start building layers, there may be mutual dependencies to worry about and other tricky build issues.

@jhpedemonte talked about manually creating groups of code that may be related or likely to be used together as a sort of compromise solution

**Comment by aerwin** [comment]
@peller 

I'm not proposing we build one giant layer. But, as I observed doing the gridx build stuff, even without a giant layer we still gain benefits from doing the bare bones dojo build of the JS parts in each plug-in (e.g., minification, inlining HTML templates, etc.). 

So, I wanted to leverage what I did for gridx so that each UI plugin could have their own basic dojo build profile and a minimalistic build would be performed for each plug-in. That's what I've put in place in the 2390_gridx_custom_build branch with commit 55ccdad8596e784bda9139a54df081a94a0fad5e . (I think this is safe to merge into master. But, when I did my test build against the server repository earlier today, I'm seeing the proxy errors that @JonFerraiolo reported yesterday. So, I want to wait for the resolution of that.)

I believe the work I've done is a stepping stone to #1152 (which I referred in the issue description above and I think you're referring to when you allude to @jhpedemonte 's proposal). For #1152, the simple dojo profiles I've put in place could be augmented to build bigger, more complicated layers. And, I think my widgetPluginDojoBuild.xml build script (now reused by each UI plugin in the 2390_gridx_custom_build branch) could be augmented to do other things alluded  in #1152 (e.g., merging json metadata files, etc.).
